### PR TITLE
Array - fold array

### PR DESCRIPTION
**Pull request by DevratAgarwal**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [x] Author
    - [x] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [x] Author
    - [x] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [x] Author
    - [x] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [x] Author
    - [x] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [x] There are at least 2 test cases
    - [x] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

**Review by rpapavan97 — COMMENTED** [review]
@DevratAgarwal 
Everything is fine, but is it possible to reduce the indentation space number to 2, so that we have same indentation across the whole code along with template

File: solutions/arrays/fold-array.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+data = data.split('\n');
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+// Use readLine() for taking input, it will read one line of from the input  and is stored in string format
+let arraySize=parseInt(readLine())
```
Comment (rpapavan97) [inline solutions/arrays/fold-array.js:13]: as mentioned in coding cconventions, please give spaces around binaru operators "=", for below line of codes

**Review by rpapavan97 — APPROVED** [review]
updated coding conventions to code (like ";" at the end of each line and giving spaces around binary operators)

**Comment by sailesh-ramayanam** [comment]
@rpapavan97 Problem description states limits for `n` and `m`. Please remove the limits. Some students give unnecessary importance to the limits.

**Comment by rpapavan97** [comment]
Modified the problem statement,  removed the limits for n and m

**Comment by DevratAgarwal** [comment]
@sailesh-ramayanam @rpapavan97 changes are addressed

**Comment by prathyusha10x** [comment]
@rpapavan97 @sailesh-ramayanam please review the updated code and approve the PR by today EOD

**Review by sailesh-ramayanam — APPROVED** [review]

**Pull request merged**